### ISSUE TITLE
Duplicate lint when another trait's supertrait is removed because of trait being removed

### ISSUE THREAD

**Issue by pksunkara** [obi1kenobi/cargo-semver-checks#604]
### Steps to reproduce the bug with the above code

Old crate:

```rust
trait A {}

trait B: A {}
```

New crate:

```rust
trait B {}
```

### Actual Behaviour

We get 2 lints:

1. `trait_missing` about `A`
2. `trait_removed_supertrait` about `B` not having `A`

### Expected Behaviour

We get 1 lint:

1. `trait_missing` about `A`

### Generated System Information

v0.26 on Rust 1.73

**Comment by obi1kenobi** [comment]
In our current approach, this is intended behavior because this is two changes:
```diff
// this causes one lint
- trait A {}

// this causes the other lint, which may be in a completely different file in the codebase
- trait B: A {}
+ trait B {}
```

We generally want to avoid multiple lints on the same item, like "you removed this struct, and also you removed each of its pub fields too." This is on multiple items that may be across multiple files, so this seems fine to me.

Do you feel there's a UX reason to change this?

**Comment by pksunkara** [comment]
IMHO, when warning that a struct is removed, warning that the fields of the struct are also removed feels like extra noise since the first message implies the second.

**Comment by obi1kenobi** [comment]
I agree:
> We generally want to avoid multiple lints on the same item, like "you removed this struct, and also you removed each of its pub fields too."

I was making the point that removal of a supertrait + removal of the trait itself is unlike that case, since it is two separate changes that may even be in different files.

**Comment by pksunkara** [comment]
> I was making the point that removal of a supertrait + removal of the trait itself is unlike that case, since it is two separate changes that may even be in different files.

Hmm.. yeah, but I was saying that if trait `A` is removed, the supertrait would need to be removed too, which means that they are not independent changes. Of course, if the trait `A` is not removed, the supertrait change would be a valid independent change.

To confirm, is your point that since they are manually changed in different places, we would want to lint them separately? I am coming from a dependency angle and not code changes.
